manim/config.py

```diff
@@ -1,283 +1,406 @@
-    # scene has a background opacity of 0
-    if args.transparent:
-        camera_config["background_opacity"] = 0
-
-    return camera_config
+    # As before, make FILE_WRITER_CONFIG by pulling form the overall
+    config['FILE_WRITER_CONFIG'] = {key: config[key] for key in [
+        "WRITE_TO_MOVIE", "SAVE_LAST_FRAME", "SAVE_PNGS",
+        "SAVE_AS_GIF", "PNG_MODE", "MOVIE_FILE_EXTENSION",
+        "OUTPUT_FILE", "INPUT_FILE"]}
 
+    return config
 
-def initialize_directories(config):
-    dir_config = {}
-    dir_config["media_dir"] = config["media_dir"] or dirs.MEDIA_DIR
-    dir_config["video_dir"] = config["video_dir"] or dirs.VIDEO_DIR
 
-    if not (config["video_dir"] and config["tex_dir"]):
-        if config["media_dir"]:
-            if not os.path.isdir(dir_config["media_dir"]):
-                os.makedirs(dir_config["media_dir"])
-        if not os.path.isdir(dir_config["media_dir"]):
-            dir_config["media_dir"] = "./media"
+def _init_dirs(config):
+    if not (config["VIDEO_DIR"] and config["TEX_DIR"]):
+        if config["MEDIA_DIR"]:
+            if not os.path.isdir(config["MEDIA_DIR"]):
+                os.makedirs(config["MEDIA_DIR"])
+        if not os.path.isdir(config["MEDIA_DIR"]):
+            config["MEDIA_DIR"] = "./media"
         else:
             print(
-                f"Media will be written to {dir_config['media_dir'] + os.sep}. You can change "
-                "this behavior with the --media_dir flag, or by adjusting dirs.py.,"
+                f"Media will be written to {config['media_dir'] + os.sep}. You can change "
+                "this behavior with the --media_dir flag, or by adjusting manim.cfg."
             )
     else:
-        if config["media_dir"]:
+        if config["MEDIA_DIR"]:
             print(
                 "Ignoring --media_dir, since both --tex_dir and --video_dir were passed."
             )
 
-    dir_config["tex_dir"] = (config["tex_dir"]
-                             or dirs.TEX_DIR
-                             or os.path.join(dir_config["media_dir"], "Tex"))
-    dir_config["text_dir"] = (config["text_dir"]
-                              or dirs.TEXT_DIR
-                              or os.path.join(dir_config["media_dir"], "texts"))
+    # Make sure all folders exist
+    for folder in [config["VIDEO_DIR"], config["TEX_DIR"], config["TEXT_DIR"]]:
+        if not os.path.exists(folder):
+            os.makedirs(folder)
 
-    if not config["video_dir"] or dirs.VIDEO_DIR:
-        dir_config["video_dir"] = os.path.join(dir_config["media_dir"], "videos")
 
-    for folder in [dir_config["video_dir"], dir_config["tex_dir"], dir_config["text_dir"]]:
-        if folder != "" and not os.path.exists(folder):
-            os.makedirs(folder)
+config_filename = 'manim.cfg'
+
+foobar =  os.path.join(os.path.dirname(__file__), config_filename)
+print(foobar)
+config_files = [
+    foobar,
+    os.path.expanduser('~/.{}'.format(config_filename)),
```

Comment:
just realized I left the foobar in -- woops

Reply:
I don't think Windows conventionally keeps config files in user's home directories. I asked @yoshiask about this and he thought `C:\Users\{user}\AppData\Local\manim` was standard.

Reply:
That's what is technically recommended, but a lot of apps do put configurations in the user's Documents directory (usually cross-platform apps). The problem is that a good portion of users don't like it when programs put stuff in their Documents.

Reply:
Ooooh yeah good point. OK so the current implementation was designed so that different config files would override the previous one in a cascading sort of way. The file in the current directory overrides the one in the user's home, which in turn overrides the library-wide defaults.

So instead of that, we could have just two files. A library-wide one, and a user-defined one that overrides the defaults. Advantage: we circumvent the issue of choosing where to place the config files. Disadvantage: if we go this way, then `manim <script>` will only ever run the library-wide defaults. Long CLI calls could be replaced by `manim <script> --config-file <config>`. What do you think?

Reply:
I like the idea of the config files cascading, so I'd rather we keep that.

Since the problem of where to keep config files could get hairy I think it'd be best to always read the global defaults, then overwrite with a local (./manim.cfg)/flag-specified config if one exists, then overwrite with flags.

Reply:
Latest commit keeps the cascading files.